concurrent_test.go

```diff
@@ -321,3 +384,100 @@ func testResultsDirectory(t *testing.T) string {
 
 	return path
 }
+
+/*
+*********************************************************
+Data structure and functions for analyzing history records
+*********************************************************
+*/
+type workerType string
+
+const (
+	reader workerType = "reader"
+	writer workerType = "writer"
+)
+
+type historyRecord struct {
+	WorkerType workerType `json:"workerType,omitempty"`
```

Comment:
ClientId? Would recommend assigning unique id to each worker rather than grouping them by type. Types can change.